repository_service_tuf_worker/services/keyvault/local.py

```diff
@@ -17,103 +21,196 @@
+                ValueError,
+                UnsupportedLibraryError,
+                StorageError,
+                Error,
+            ) as e:
+                logging.error(str(e))
+                logging.warning("Failed to load LocalKeyVault key")
+
+        if valid_key_found is False:
+            error = KeyVaultError("No valid keys found in the LocalKeyVault")
+            logging.error("No valid keys found in the LocalKeyVault")
+            raise error
 
     @classmethod
     def settings(cls) -> List[ServiceSettings]:
         """Define the settings parameters."""
         return [
             ServiceSettings(
                 name="LOCAL_KEYVAULT_PATH",
-                argument="key_path",
-                required=False,
-                default="online.key",
-            ),
-            ServiceSettings(
-                name="LOCAL_KEYVAULT_PASSWORD",
-                argument="key_pass",
+                argument="path",
                 required=True,
             ),
             ServiceSettings(
-                name="LOCAL_KEYVAULT_TYPE",
-                argument="key_type",
-                required=False,
-                default="ed25519",
+                name="LOCAL_KEYVAULT_KEYS",
+                argument="keys",
+                required=True,
             ),
         ]
 
+    def _secrets_handler(self, password: str) -> Callable:
+        """Generates simple Callable with password - required by SSLibSigner"""
+        return lambda *a: password
+
     def get(self, public_key: Key) -> SSlibSigner:
         """Return a signer using the online key."""
-        try:
-            priv_key_uri = f"file:{self._path}?encrypted=true"
-            sslib_public_key = SSlibKey.from_dict(
-                public_key.keyid, public_key.to_dict()
-            )
-            return SSlibSigner.from_priv_key_uri(
-                priv_key_uri, sslib_public_key, self._secrets_handler
-            )
-        except ValueError as e:
-            raise KeyVaultError("Cannot load the online key") from e
-        except OSError:
-            raise KeyVaultError(f"Cannot read private key file {self._path}")
+        keys = self._raw_key_parser(self._path, self._keys)
+        valid_key = False
+        for key in keys:
+            key_path = os.path.join(self._path, key.file)
+            priv_key_uri = f"file:{key_path}?encrypted=true"
+            try:
+                sslib_public_key = SSlibKey.from_dict(
+                    public_key.keyid, public_key.to_dict()
+                )
+            except ValueError:
+                logging.error("Cannot load the online key")
+                continue
+            except OSError:
+                logging.error("Cannot read private key")
+                continue
+
+            try:
+                sslib_signer = SSlibSigner.from_priv_key_uri(
+                    priv_key_uri,
+                    sslib_public_key,
+                    self._secrets_handler(key.password),
```

Comment:
I call because the `_secrets_handler` returns a Callable object (`lambda`)
The `_secrets_handler` helps to create a persistent object in the tests instead of using the `lambda`

Reply:
Oh, you have a point, you are correct.